common/components/org.wso2.carbon.data.agents.log4j/src/main/java/org/wso2/carbon/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -0,0 +1,411 @@
+                            getMaxTolerableConsecutiveFailure() + ". Purging log event array. Some logs will be lost.");
+                    loggingEvents.clear();
+                    numOfConsecutiveFailures = 0;
+                }
+            }
+        }
+
+        private void publishLogEvent(TenantAwareLoggingEvent event)
+                throws ParseException, MalformedURLException, AuthenticationException, TransportException,
+                MalformedStreamDefinitionException, StreamDefinitionException,
+                DifferentStreamDefinitionAlreadyDefinedException, ExecutionException {
+            String streamId = "";
+            String streamName = "";
+            String tenantId = event.getTenantId();
+            if (tenantId.equals(String.valueOf(MultitenantConstants.INVALID_TENANT_ID)) || tenantId
+                    .equals(String.valueOf(MultitenantConstants.SUPER_TENANT_ID))) {
+                tenantId = "0";
+            }
+            String serverKey = getCurrentServerName();
+            String currDateStr = getCurrentDate();
+            try {
+                Pattern pattern = Pattern.compile("(tcp):\\/\\/([a-zA-Z0-9]+):([0-9]+)");
+                Matcher matcher = pattern.matcher(url);
+                if (matcher.find() && authURLSet == null) {
+                    authURLSet = "ssl://" + matcher.group(2).toString() + ":" + (
+                            Integer.parseInt(matcher.group(3).toString()) + 100);
+                }
+                dataPublisher = new DataPublisher("Thrift", url, authURLSet, userName, password);
+            } catch (DataEndpointAgentConfigurationException e) {
+                log.error(
+                        "Error in initializing binary data-publisher to send requests to global throttling engine " + e
+                                .getMessage(), e);
+            } catch (DataEndpointException e) {
+                log.error(
+                        "Error in initializing binary data-publisher to send requests to global throttling engine " + e
+                                .getMessage(), e);
+            } catch (DataEndpointConfigurationException e) {
+                log.error(
+                        "Error in initializing binary data-publisher to send requests to global throttling engine " + e
+                                .getMessage(), e);
+            } catch (DataEndpointAuthenticationException e) {
+                log.error(
+                        "Error in initializing binary data-publisher to send requests to global throttling engine " + e
+                                .getMessage(), e);
+            } catch (TransportException e) {
+                log.error(
+                        "Error in initializing binary data-publisher to send requests to global throttling engine " + e
+                                .getMessage(), e);
+            }
```

Comment:
Could you explain exception handling logic here? What happens after exception(s) handled here? Do you exit or do you still continue?